src/ReplayPlugin.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Http\Client\Plugin\Vcr;
+
+use Http\Client\Common\Plugin;
+use Http\Client\Plugin\Vcr\NamingStrategy\NamingStrategyInterface;
+use Http\Client\Plugin\Vcr\Recorder\PlayerInterface;
+use Http\Promise\FulfilledPromise;
+use Http\Promise\Promise;
+use Psr\Http\Message\RequestInterface;
+
+final class ReplayPlugin implements Plugin
+{
+    const HEADER_NAME = 'X-VCR-REPLAYED';
+
+    /**
+     * @var NamingStrategyInterface
+     */
+    private $namingStrategy;
+
+    /**
+     * @var PlayerInterface
+     */
+    private $player;
+
+    public function __construct(NamingStrategyInterface $namingStrategy, PlayerInterface $player)
+    {
+        $this->namingStrategy = $namingStrategy;
+        $this->player = $player;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handleRequest(RequestInterface $request, callable $next, callable $first): Promise
+    {
+        $name = $this->namingStrategy->name($request);
+
+        if ($response = $this->player->replay($name)) {
+            return new FulfilledPromise($response->withAddedHeader(static::HEADER_NAME, $name));
+        }
+
+        return $next($request);
```

Comment:
i think we should throw an exception in this case. in most CI settings, you want to know when your fixture has not been found.

Reply:
Yup, but that would prevent a `dev` replay OR record scenario. Maybe an option on the ReplayPlugin?

Reply:
yep, that sounds good to me. i'd default to error, so people consiously decide they want fallback, rather than not notice what they do.

Reply:
Done!

Reply:
perfect